### PR TITLE
Check for a reset before applying preferences

### PR DESCRIPTION
**Pull request by rak-phillip**
This proposes preferences and prompts the user if a reset is required before applying preferences. 

contributes to the critical path for #2441

### CODE REVIEW

File: src/store/preferences.ts [hunk 1]
```diff
@@ -112,6 +124,32 @@ export const actions = {
...
+
+    const severities = Object.entries(await result.json())
+      .map(([_key, val]: [key: string, val: any]) => {
+        const { severity } = val;
+
+        return { severity };
+      })
+      .reduce((prev, curr) => {
+        return { ...prev, [curr.severity]: true };
+      }, { reset: false, restart: false });
```
Comment (mook-as) [inline src/store/preferences.ts:148]: Testing out alternatives…
```suggestion
    const changes: Record<string, {severity: 'reset' | 'restart'}> = await result.json();
    const values = Object.values(changes).map(v => v.severity);
    const severities = {
      reset:   values.includes('reset'),
      restart: values.includes('restart'),
    };
```
I'm not actually sure the alternative is better; maybe a combination of the two?

**Review by mook-as — DISMISSED** [review]
It would be good to show a warning _before_ the user clicks the Apply button (probably to the left of the Apply/Cancel buttons).  Ideally, each element that causes a reset/restart would be labelled.  (Both of these things can happen in follow-up issues.)

Approving with comments — if you like your existing code over the suggestion, feel free to just merge directly.

**Comment by jandubois** [comment]
> It would be good to show a warning before the user clicks the Apply button (probably to the left of the Apply/Cancel buttons).

That was the idea behind having the "dry-run" API so you could show this as soon as the user makes a change that requires a restart of either kubernetes or the VM.

We can always add this later, if we are running out of time now.

File: src/store/preferences.ts (hunk 1 above)
Comment (rak-phillip) [reply src/store/preferences.ts]: I think your recommendation is quite elegant and easy to read, thanks 🙂

**Comment by rak-phillip** [comment]
@mook-as @jandubois FYI I framed this pr using the wording "contributes" instead of using any closing keywords for this very reason. We get our core behavior that we outlined here and can more than easily expand on what we have in a follow-up PR

**Review by mook-as — APPROVED** [review]

**Pull request merged**